cardano-node/src/Cardano/Node/Tracing/Tracers/ChainDB.hs

```diff
@@ -0,0 +1,1761 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+{-# OPTIONS_GHC -Wno-orphans  #-}
+
+module Cardano.Node.Tracing.Tracers.ChainDB
+  ( severityChainDB
+  , namesForChainDBTraceEvents
+  , withAddedToCurrentChainEmptyLimited
+  , docChainDBTraceEvent
+  ) where
+
+import           Data.Aeson (Value (String), toJSON, (.=))
+import qualified Data.Aeson as A
+import           Data.HashMap.Strict (insertWith)
+import qualified Data.Text as Text
+import qualified Data.Text.Encoding as Text
+import           Numeric (showFFloat)
+import           Prelude (id)
+import           Text.Show
+
+import           Cardano.Logging
+import           Cardano.Node.Tracing.Era.Byron ()
+import           Cardano.Node.Tracing.Era.Shelley ()
+import           Cardano.Node.Tracing.Formatting ()
+import           Cardano.Node.Tracing.Render
+import           Cardano.Prelude hiding (Show, show, trace)
+
+import           Ouroboros.Consensus.Block
+import           Ouroboros.Consensus.HeaderValidation (HeaderEnvelopeError (..), HeaderError (..),
+                   OtherHeaderEnvelopeError)
+import           Ouroboros.Consensus.Ledger.Abstract (LedgerError)
+import           Ouroboros.Consensus.Ledger.Extended (ExtValidationError (..))
+import           Ouroboros.Consensus.Ledger.Inspect (InspectLedger, LedgerEvent (..))
+import           Ouroboros.Consensus.Ledger.SupportsProtocol (LedgerSupportsProtocol)
+import           Ouroboros.Consensus.Protocol.Abstract (ValidationErr)
+import qualified Ouroboros.Consensus.Protocol.PBFT as PBFT
+import qualified Ouroboros.Consensus.Storage.ChainDB as ChainDB
+import qualified Ouroboros.Consensus.Storage.ImmutableDB as ImmDB
+import           Ouroboros.Consensus.Storage.ImmutableDB.Chunks.Internal (chunkNoToInt)
+import qualified Ouroboros.Consensus.Storage.ImmutableDB.Impl.Types as ImmDB
+import qualified Ouroboros.Consensus.Storage.LedgerDB.OnDisk as LedgerDB
+import           Ouroboros.Consensus.Storage.LedgerDB.Types (UpdateLedgerDbTraceEvent (..))
+import qualified Ouroboros.Consensus.Storage.LedgerDB.Types as LedgerDB
+import qualified Ouroboros.Consensus.Storage.VolatileDB as VolDB
+import qualified Ouroboros.Consensus.Storage.VolatileDB.Impl as VolDb
+import           Ouroboros.Consensus.Util.Condense (condense)
+
+import qualified Ouroboros.Network.AnchoredFragment as AF
+
+{-# ANN module ("HLint: ignore Redundant bracket" :: Text) #-}
+
+withAddedToCurrentChainEmptyLimited ::
```

Comment:
Should be `:: Trace IO (ChainDB.TraceEvent blk)`

Reply:
Done